### ISSUE TITLE
Framework: esp-idf not compatible?

### ISSUE THREAD

**Issue by dafunkydan** [MrUndead1996/ld2410s-esphome#2]
Thank you so much for this component!
I just got some LD2410S after i found out you did that component 👍 

When using framework: arduino, i can sucessfully compile it. However, i need to use esp-idf.
Then i get some errors when compiling, e.g.
```
Compiling .pioenvs/esp32--presence03/src/esphome/components/ld2410s/button/LD2410SButtons.o
src/esphome/components/ld2410s/LD2410S.cpp: In member function 'void esphome::ld2410s::LD2410S::process_data_package(esphome::ld2410s::PackageType)':
src/esphome/components/ld2410s/LD2410S.cpp:192:20: error: enumeration value 'ACK' not handled in switch [-Werror=switch]
             switch (type)
                    ^
src/esphome/components/ld2410s/LD2410S.cpp:192:20: error: enumeration value 'UNKNOWN' not handled in switch [-Werror=switch]
Compiling .pioenvs/esp32--presence03/src/esphome/components/ld2410s/number/LD2410SNumber.o
cc1plus: some warnings being treated as errors
*** [.pioenvs/esp32--presence03/src/esphome/components/ld2410s/LD2410S.o] Error 1
```

Any chance to get it compatible to esp-idf?

Thanks again for your Work, hope i can give it a shot soon 😎

**Comment by MrUndead1996** [comment]
Hello.

Support for esp-idf framework was added

**Comment by dafunkydan** [comment]
Lightning fast, Thanks!
Unfortunatly, exactly same Error comes up :-(

I doubt it helps, but here the full Log:

```
INFO ESPHome 2024.10.2
INFO Reading configuration /config/esphome/esp32--presence03.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing esp32--presence03 (board: esp32dev; framework: espidf; platform: platformio/espressif32@5.4.0)
--------------------------------------------------------------------------------
Library Manager: Installing esphome/noise-c @ 0.1.6
INFO Installing esphome/noise-c @ 0.1.6
Unpacking  [####################################]  100%
Library Manager: noise-c@0.1.6 has been installed!
INFO noise-c@0.1.6 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/libsodium @ 1.10018.4
INFO Installing esphome/libsodium @ 1.10018.4
Unpacking  [####################################]  100%
Library Manager: libsodium@1.10018.4 has been installed!
INFO libsodium@1.10018.4 has been installed!
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - framework-espidf @ 3.40408.0 (4.4.8) 
 - tool-cmake @ 3.16.9 
 - tool-ninja @ 1.10.2 
 - toolchain-esp32ulp @ 2.35.0-20220830 
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Reading CMake configuration...
Generating assembly for certificate bundle...
Dependency Graph
|-- noise-c @ 0.1.6
Compiling .pioenvs/esp32--presence03/src/esphome/components/adc/adc_sensor.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/api/api_connection.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/api/api_frame_helper.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/api/api_pb2.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/api/api_pb2_service.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/api/api_server.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/api/list_entities.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/api/proto.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/api/subscribe_state.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/api/user_services.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/binary_sensor/automation.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/binary_sensor/binary_sensor.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/binary_sensor/filter.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/button/button.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/captive_portal/captive_portal.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/dht/dht.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/esp32/core.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/esp32/gpio.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/esp32/preferences.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/esp32_ble/ble.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/esp32_ble/ble_advertising.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/esp32_ble/ble_uuid.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/esp32_ble_tracker/esp32_ble_tracker.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/esp32_touch/esp32_touch.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/esphome/ota/ota_esphome.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/gpio/binary_sensor/gpio_binary_sensor.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/gpio/switch/gpio_switch.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/homeassistant/sensor/homeassistant_sensor.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/internal_temperature/internal_temperature.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/ld2410s/LD2410S.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/ld2410s/button/LD2410SButtons.o
Compiling .pioenvs/esp32--presence03/src/esphome/components/ld2410s/number/LD2410SNumber.o
src/esphome/components/ld2410s/LD2410S.cpp: In member function 'void esphome::ld2410s::LD2410S::process_data_package(esphome::ld2410s::PackageType)':
src/esphome/components/ld2410s/LD2410S.cpp:192:20: error: enumeration value 'ACK' not handled in switch [-Werror=switch]
             switch (type)
                    ^
src/esphome/components/ld2410s/LD2410S.cpp:192:20: error: enumeration value 'UNKNOWN' not handled in switch [-Werror=switch]
cc1plus: some warnings being treated as errors
Compiling .pioenvs/esp32--presence03/src/esphome/components/ld2410s/select/LD2420ResponseSpeedSelect.o
*** [.pioenvs/esp32--presence03/src/esphome/components/ld2410s/LD2410S.o] Error 1
```

I cleaned the Build Files before compiling - anything more i can try?

**Comment by MrUndead1996** [comment]
I see same error. Did you update component source code?

**Comment by dafunkydan** [comment]
I linked it directly to your component:

```
external_components:
  - source:
      type: git
      url: https://github.com/MrUndead1996/ld2410s-esphome
```

**Comment by MrUndead1996** [comment]
I think you need to reduce refresh parameter, because default value is 1 day

**Comment by dafunkydan** [comment]
Cant neither find refresh parameters for esphome, nor cached files anywhere. Most probably i am just too blind.
Will give it a shot tomorrow, probably just downloading the component and testing it local.
Thanks for your assistance - and your component!

**Comment by MrUndead1996** [comment]
It described [here](https://esphome.io/components/external_components#external-components) on the end of page.
And [here](https://esphome.io/components/external_components#external-components) upper than **Local** section

TL;DR

```
 external_components:
  - source:
      type: git
      url: https://github.com/MrUndead1996/ld2410s-esphome
      refresh: 0s
```

**Comment by dafunkydan** [comment]
Thanks! Just as you posted it saw it myself 😄 Will check tomorrow asap! Thanks, highly appreciate it!

**Comment by dafunkydan** [comment]
Sure, that did the Trick. Blind me.... Thank you so much! 
Works great so far, awesome that i can tinker with that great little Module now, appreciate it! 👍
